Automattic/wp-calypso
# Inline Help: Remove Checklist prompt

#### Changes proposed in this Pull Request

* I giveth, and I taketh away. 😂 Removes the checklist "prompt" feature from inline help. Now that My Home is the default landing page, it's redundant.
* Removes checklist markup from the Checklist component itself.
* Removes related redux values for showing the checklist notification and tasks.

**Before**

<img width="352" alt="Screen Shot 2020-03-24 at 3 50 27 PM" src="https://user-images.githubusercontent.com/2124984/77470760-92716780-6de7-11ea-9248-0743aecabff4.png">

**After**

<img width="351" alt="Screen Shot 2020-03-24 at 3 50 11 PM" src="https://user-images.githubusercontent.com/2124984/77470767-969d8500-6de7-11ea-9edf-7b106c8a3e91.png">

#### Testing instructions

* Switch to this PR
* Navigate around Calypso and confirm the checklist prompt does not appear in Inline Help
* Make sure there are no errors in the console as a result of the removed code

Fixes #40384
